src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -212,13 +199,25 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
         /// </summary>
         private string? GetFinalUri(Response response)
         {
-            // Set final uri as null if the response header doesn't contain "Operation-Location" or "Azure-AsyncOperation".
+            // Set final uri as null if the response for initial request doesn't contain header "Operation-Location" or "Azure-AsyncOperation".
             if (_headerSource is not (HeaderSource.OperationLocation or HeaderSource.AzureAsyncOperation))
             {
                 return null;
             }
 
-            if (_finalStateVia == OperationFinalStateVia.AzureAsyncOperation && _requestMethod == RequestMethod.Post)
+            // Handle final-state-via options: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation-options
+            switch (_finalStateVia)
+            {
+                case OperationFinalStateVia.Location when _originalResponseHasLocation:
+                    return _lastKnownLocation;
+                case OperationFinalStateVia.OperationLocation or OperationFinalStateVia.AzureAsyncOperation:
+                    return null;
+                case OperationFinalStateVia.OriginalUri:
+                    return _startRequestUri.AbsoluteUri;
```

Comment:
So rather than check for PUT- and PATCH-based LROs like management-plane use as well as some data-plane services like Cognitive Services - Question Answering, we'll have to explicit set them to `OriginalUri`? How will that be determined in the generator? If opt in, what happens to existing SDKs including management-plane SDKs? /cc @m-nash

Reply:
For QnA, you don't need to specify anything for `OperationFinalStateVia`. Since now there will be default value `OperationFinalStateVia.NotSpecified`, it will fall through to the PUT and PATCH

Reply:
Talking offline with you and @m-nash I'm not comfortable this won't regress other services. Would it be enough to just transform in the `x-ms-long-running-operation-options.final-state-via` to `original-uri` in my SDK for now?